File: Cmdlets/Cmdlets/GetRegionInfoCommand.vb
<System.Management.Automation.Cmdlet("Get", "RegionInfo")> _
Public Class GetRegionInfoCommand
    Inherits System.Management.Automation.Cmdlet

    Public Shared Function GetAllRegions(Optional CultureOnly As Boolean = False) As System.Collections.Generic.List(Of System.Globalization.RegionInfo)
        Dim RegionNames As New System.Collections.Generic.HashSet(Of String)

        For Each CI In System.Globalization.CultureInfo.GetCultures(System.Globalization.CultureTypes.SpecificCultures)
            RegionNames.Add(CI.Name)
        Next

        Dim Regions As New System.Collections.Generic.List(Of System.Globalization.RegionInfo)

        For Each RegionName In RegionNames
            Regions.Add(New System.Globalization.RegionInfo(RegionName))
        Next

        If Not CultureOnly Then
            For Each Region In Regions
                RegionNames.Add(Region.TwoLetterISORegionName)
            Next

            Regions.Clear()

            For Each RegionName In RegionNames
                Regions.Add(New System.Globalization.RegionInfo(RegionName))
            Next
        End If

        Return Regions
    End Function

    <System.Management.Automation.Parameter(Mandatory:=False)> _
    Public Property CultureOnly() As System.Management.Automation.SwitchParameter

    <System.Management.Automation.Parameter(Mandatory:=False)> _
    Public Property Name As String

    <System.Management.Automation.Parameter()> _
    Public Property All() As System.Management.Automation.SwitchParameter

    Protected Overrides Sub BeginProcessing()

        
        If Not String.IsNullOrEmpty(Me.Name) Then

            Dim Region = New System.Globalization.RegionInfo(Me.Name)

            Me.WriteObject(Region)
        ElseIf Me.All.IsPresent OrElse Me.CultureOnly.IsPresent Then

            Dim Regions = GetAllRegions(Me.CultureOnly.IsPresent)

            Me.WriteObject(Regions, True)
        Else

            Me.WriteObject(System.Globalization.RegionInfo.CurrentRegion)
        End If
    End Sub
End Class

File: Cmdlets.Core/Commands/GetRegionInfoCommand.cs
using TIKSN.Cmdlets.Common.Helpers;

namespace TIKSN.Cmdlets.Core.Commands
{
    [System.Management.Automation.Cmdlet("Get", "RegionInfo")]
    public class GetRegionInfoCommand : System.Management.Automation.Cmdlet
    {
        [System.Management.Automation.Parameter()]
        public System.Management.Automation.SwitchParameter All { get; set; }

        [System.Management.Automation.Parameter(Mandatory = false)]
        public System.Management.Automation.SwitchParameter CultureOnly { get; set; }

        [System.Management.Automation.Parameter(Mandatory = false)]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            if (!string.IsNullOrEmpty(this.Name))
            {
                var Region = new System.Globalization.RegionInfo(this.Name);

                this.WriteObject(Region);
            }
            else if (this.All.IsPresent || this.CultureOnly.IsPresent)
            {
                var Regions = RegionHelper.GetAllRegions(this.CultureOnly.IsPresent);

                this.WriteObject(Regions, true);
            }
            else
            {
                this.WriteObject(System.Globalization.RegionInfo.CurrentRegion);
            }
        }
    }
}